src/test/java/org/csanchez/jenkins/plugins/kubernetes/PodTemplateMapTest.java

```diff
@@ -39,7 +42,7 @@ public void concurrentAdds() throws Exception {
         for (Thread thread : t) {
             thread.join();
         }
-        assertEquals(n, this.instance.getTemplates(cloud).size());
+        Awaitility.await().atMost(5, TimeUnit.SECONDS).until(() -> this.instance.getTemplates(cloud).size() == n);
```

Comment:
Could you add a code comment as to why this is needed? (where is the async nature coming from?)